test/Pester/EditorServices.Integration.Tests.ps1

```diff
@@ -0,0 +1,78 @@
+
+$script:ExceptionRegex = [regex]::new('\s*Exception: (.*)$', 'Compiled,Multiline,IgnoreCase')
+function ReportLogErrors
+{
+    param(
+        [Parameter()][string]$LogPath,
+
+        [Parameter()][ref]<#[int]#>$FromIndex = 0,
+
+        [Parameter()][string[]]$IgnoreException = @()
+    )
+
+    $logEntries = Parse-PsesLog $LogPath |
+        Where-Object Index -ge $FromIndex.Value
+
+    # Update the index to the latest in the log
+    $FromIndex.Value = ($FromIndex.Value,$errorLogs.Index | Measure-Object -Maximum).Maximum
+
+    $errorLogs = $logEntries |
+        Where-Object LogLevel -eq Error |
+        Where-Object {
+            $match = $script:ExceptionRegex.Match($_.Message.Data)
+
+            (-not $match) -or ($match.Groups[1].Value.Trim() -notin $IgnoreException)
+        }
+
+    if ($errorLogs)
+    {
+        $errorLogs | ForEach-Object { Write-Error "ERROR from PSES log: $($_.Message.Data)" }
+    }
+}
+
+Describe "Loading and running PowerShellEditorServices" {
+    BeforeAll {
+        Import-Module -Force "$PSScriptRoot/../../module/PowerShellEditorServices"
+        Import-Module -Force "$PSScriptRoot/../../tools/PsesPsClient/out/PsesPsClient"
+        Import-Module -Force "$PSScriptRoot/../../tools/PsesLogAnalyzer"
+
+        $logIdx = 0
+        $psesServer = Start-PsesServer
+        $client = Connect-PsesServer -PipeName $psesServer.SessionDetails.languageServicePipeName
+    }
+
+    AfterAll {
+        try
+        {
+            $client.Dispose()
+            $psesServer.PsesProcess.Kill()
+            $psesServer.PsesProcess.Dispose()
+        }
+        catch
+        {
+            # Do nothing
+        }
+
+        # TODO: We shouldn't need to skip this error
```

Comment:
Why does this exception happen? Can you add a comment?

Reply:
Turns out disposing the client before the server is not what the server expects